x-pack/plugins/stack_connectors/public/connector_types/lib/validate_json.test.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { validateJSON } from './validate_json';
+
+describe('validateJSON', () => {
+  it('does not return an error for valid JSON and no maxProperties', () => {
+    expect(validateJSON({ value: JSON.stringify({ foo: 'test' }) })).toBeUndefined();
+  });
+
+  it('does not return an error for valid JSON and attributes less than maxProperties', () => {
+    expect(
+      validateJSON({ value: JSON.stringify({ foo: 'test' }), maxProperties: 1 })
+    ).toBeUndefined();
+  });
+
+  it('does not return an error with empty value and maxProperties=0', () => {
+    expect(validateJSON({ maxProperties: 0 })).toBeUndefined();
```

Comment:
If the value is empty is undefined it wont ever get to the `maxProperties` check in `validateJSON`.

How about:
- one test where we do `validateJSON({})`
- and one `validateJSON({ value: JSON.stringify({}), maxProperties: 0 })`

Reply:
Good point!